pilot/pkg/model/service.go

```diff
@@ -245,10 +245,11 @@ func (instance *ServiceInstance) DeepCopy() *ServiceInstance {
 }
 
 type WorkloadInstance struct {
-	Name      string            `json:"name,omitempty"`
-	Namespace string            `json:"namespace,omitempty"`
-	Endpoint  *IstioEndpoint    `json:"endpoint,omitempty"`
-	PortMap   map[string]uint32 `json:"portMap,omitempty"`
+	Name           string            `json:"name,omitempty"`
+	Namespace      string            `json:"namespace,omitempty"`
+	Endpoint       *IstioEndpoint    `json:"endpoint,omitempty"`
+	PortMap        map[string]uint32 `json:"portMap,omitempty"`
+	InTransitState bool              `json:"inTransitState,omitempty"`
```

Comment:
Prefer to check this before calling the workload handler

Reply:
Or leave the logic inside workload Handler. Because as our previous discussion suggests, this is only used by `workloadInstanceHandler`.